tests/test-core/utils/js/ConditionalTestUtils.js

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+
+*/
+
+/* global fluid, jqUnit */
+
+(function () {
+    "use strict";
+
+    fluid.registerNamespace("fluid.test.conditionalTestUtils");
+
+    // "platformArg" is a string to check against navigator.platform
+
+    fluid.test.conditionalTestUtils.isPlatform = function (platformArg) {
+        var detectedPlatform = navigator.platform ? navigator.platform : undefined;
```

Comment:
@amb26 that function still seems to require a `that-type` reference; I think what I'm looking for (possibly incorrectly) is a way to have something like the following:

```
fluid.contextAware.getContextAwareCheckValue("fluid.platform");
```

Does that make sense, or is that an inadvisable approach?